### PR TITLE
create wrapper script in order to wait for Nextcloud to start

### PR DESCRIPTION
**Pull request by szaimen**
For https://github.com/nextcloud/all-in-one/pull/3614

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Wait for Nextcloud to come online
+if [ -n "$NC_DOMAIN" ] && [ "$NC_PORT" ]; then
```
Comment (szaimen) [inline start.sh:4]:
```suggestion
if [ -n "$NC_DOMAIN" ] && [ -n "$NC_PORT" ]; then
```

File: Dockerfile [hunk 2]
```diff
@@ -3,5 +3,17 @@ WORKDIR /app
...
+    apt-get install -y --no-install-recommends \
+        tzdata \
+        netcat-openbsd \
+    ; \
+    rm -rf /var/lib/apt/lists/*;
+
+COPY --chmod=775 start.sh /start.sh
+
 EXPOSE 8080
+ENTRYPOINT /start.sh
```
Comment (thanek) [inline Dockerfile:18]:
```suggestion
ENTRYPOINT ["/start.sh"]
```

File: start.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Wait for Nextcloud to come online
+if [ -n "$NC_DOMAIN" ] && [ -n "$NC_PORT" ]; then
+    while ! nc -z "$NC_DOMAIN" "$NC_PORT"; do
+        echo "Waiting for Nextcloud to start..."
```
Comment (thanek) [inline start.sh:6]:
```suggestion
        echo "Waiting for Nextcloud to start on $NC_DOMAIN:$NC_PORT..."
```

File: start.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+        sleep 5
+    done
```
Comment (thanek) [inline start.sh:8]:
```suggestion
    done
    echo "Nextcloud found on $NC_DOMAIN:$NC_PORT!"
```

Comment (thanek) [reply start.sh:8]: Let's add some messages that would help in debugging if any problems occur.

File: Dockerfile (hunk 2 above)
Comment (thanek) [reply Dockerfile:18]: Without this change the container failed to run (falling in restart loop)

File: start.sh (hunk 4 above)
Comment (szaimen) [reply start.sh:8]: done

File: start.sh (hunk 3 above)
Comment (szaimen) [reply start.sh]: done

File: Dockerfile (hunk 2 above)
Comment (szaimen) [reply Dockerfile]: done

**Review by thanek — APPROVED** [review]

**Pull request merged**